File: Nim/stats.nim
import sequtils, future, math, strformat

proc mean[T](vec: seq[T]): float64 =
  result = float64(vec.sum) / float64(vec.len)
  
proc variance[T](vec: seq[T]): float64 =
  let l = float64(vec.len)
  let m = vec.mean
  result = vec.map(x => (x.float64 - m)^2 / (l - 1)).sum

proc std[T](vec: seq[T]): float64 =
  vec.variance.sqrt

proc main =
  let a = toSeq 1..10000000
  echo &"Mean: {a.mean}"
  echo &"Var: {a.variance}"
  echo &"Std: {a.std}"

if isMainModule:
  main()

File: Rust/src/stats.rs
pub trait Statistics {
    fn mean(&self) -> f64;
    fn var(&self) -> f64;
    fn std(&self) -> f64;
}

pub type Vector = Vec<f64>;

impl Statistics for Vector {
    fn mean(&self) -> f64 {
        let l = self.len() as f64;
        self.into_iter().fold(0f64, |x, y| x + y) / l
    }
    fn var(&self) -> f64 {
        let l = self.len();
        let mut s: f64 = 0f64;
        let m = self.mean();
        for i in 0 .. l {
            s +=  (self[i] - m).powf(2.) / (l as f64 - 1f64);
        }
        return s;
    }
    fn std(&self) -> f64 {
        self.var().sqrt()
    }
}

